pkg/asset/manifests/vsphere/cloudproviderconfig.go

```diff
@@ -4,66 +4,34 @@ import (
 	"bytes"
 	"fmt"
 
-	"github.com/pkg/errors"
-	ini "gopkg.in/ini.v1"
-
 	vspheretypes "github.com/openshift/installer/pkg/types/vsphere"
 )
 
-type config struct {
-	Global    global
-	Workspace workspace
+func printIfNotEmpty(buf *bytes.Buffer, k, v string) {
+	if v != "" {
+		fmt.Fprintf(buf, "%s = %q\n", k, v)
+	}
 }
 
-type global struct {
-	SecretName      string `ini:"secret-name"`
-	SecretNamespace string `ini:"secret-namespace"`
-	InsecureFlag    int    `ini:"insecure-flag"`
-}
+// CloudProviderConfig generates the cloud provider config for the vSphere platform.
+func CloudProviderConfig(clusterName string, p *vspheretypes.Platform) (string, error) {
+	buf := new(bytes.Buffer)
 
-type workspace struct {
-	Server           string `ini:"server"`
-	Datacenter       string `ini:"datacenter"`
-	DefaultDatastore string `ini:"default-datastore"`
-	Folder           string `ini:"folder"`
-}
+	fmt.Fprintln(buf, "[Global]")
+	printIfNotEmpty(buf, "secret-name", "vsphere-creds")
+	printIfNotEmpty(buf, "secret-namespace", "kube-system")
+	printIfNotEmpty(buf, "insecure-flag", "1")
+	fmt.Fprintln(buf, "")
 
-type virtualCenter struct {
-	Datacenters string `ini:"datacenters"`
-}
+	fmt.Fprintln(buf, "[Workspace]")
+	printIfNotEmpty(buf, "server", p.VCenter)
+	printIfNotEmpty(buf, "datacenter", p.Datacenter)
+	printIfNotEmpty(buf, "default-datastore", p.DefaultDatastore)
+	printIfNotEmpty(buf, "folder", clusterName)
+	fmt.Fprintln(buf, "")
+
+	fmt.Fprintf(buf, "[VirtualCenter %q]\n", p.VCenter)
```

Comment:
`Fprintf` -> `Fprintln`

Reply:
there's a `%q` verb; do you mean the variable shouldn't be quoted here?

Reply:
@pierreprinetti no. I misread that. Your original code is good.